frontend/src/components/App/Notifications/index.tsx

```diff
@@ -0,0 +1,226 @@
+    if (!notification) {
+      return;
+    }
+    notification.seen = true;
+    clickEventHandler(notification);
+  }
+
+  function notificationItemClickHandler(notification: Notification) {
+    notification.url && history.push(notification.url);
+    notification.seen = true;
+    clickEventHandler(notification, true);
+  }
+
+  function Row(props: { index: number; style: object }) {
+    const { index, style } = props;
+    const notification = notifications[index];
+    return (
+      <MenuItem
+        key={`${notification}__${index}`}
+        className={classes.notificationItem}
+        style={style}
+      >
+        <Grid
+          container
+          justifyContent="space-between"
+          spacing={1}
+          onClick={() => notificationItemClickHandler(notification)}
+        >
+          <Grid item md={notification.seen ? 11 : 10}>
+            <Tooltip title={`${t(`notifications|${notification.message}`)}`}>
+              <Typography style={{ fontWeight: notification.seen ? 'normal' : 'bold' }} noWrap>
+                {`${notification.message || t(`notifications|No message`)}`}
+              </Typography>
+            </Tooltip>
+          </Grid>
+          {!notification.seen && (
+            <Grid item md={1}>
+              <Badge
+                variant="dot"
+                color="error"
+                onClick={e => notificationSeenUnseenHandler(e, notification)}
+              ></Badge>
+            </Grid>
+          )}
+          <Grid item md={12}>
+            <DateLabel date={notification.date} />
+          </Grid>
+        </Grid>
+      </MenuItem>
+    );
+  }
+
+  return (
+    <FixedSizeList
+      itemCount={notifications?.length}
+      height={400}
+      itemData={notifications}
+      width="100"
+      itemSize={100}
+    >
+      {Row}
+    </FixedSizeList>
+  );
+}
+
+export default function Notifications() {
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const notifications = useTypedSelector(state => state.ui.notifications);
+  const dispatch = useDispatch();
+  const [events] = Event.useList();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    if (events && events.length !== 0) {
+      const importantEvents: Notification[] = events
+        .filter((event: KubeEvent) => event.type !== 'Normal')
+        .map((event: KubeEvent) => {
+          const notificationIndexFromStore = notifications.findIndex(
+            notification => notification.id === event.metadata.uid
+          );
+          if (notificationIndexFromStore !== -1) {
+            return notifications[notificationIndexFromStore];
+          }
+          const notification = new Notification();
+          notification.date = new Date(event.metadata.creationTimestamp).getTime();
+          notification.id = event.metadata.uid;
+          notification.message = event.message;
+          notification.url = createRouteURL('cluster') + `?eventsFilter=${notification.id}`;
```

Comment:
This could be: `notification.url = createRouteURL('cluster',  {eventsFilter: notification.id});`

Reply:
Adding it as a second argument will not work for our case because we want to pass the filter as queryParam but the second argument of ```createRouteURL``` processes urlParams